### ISSUE TITLE
Add "language" as an input aspect to rules that has to do with evaluating meaning (c4a8a4, cc0f0a, b49b2e, ebe86a, 9eb3f6, 1c6c4b, b20e66, etc.)

### ISSUE THREAD

**Issue by annethyme** [act-rules/act-rules.github.io#671]
From [Accessibility Conformance Testing Rules: Common Input Aspects]( https://w3c.github.io/wcag-act/NOTE-act-rules-common-aspects.html#input-aspects-text): 

> Language, either written or spoken, contained in nodes of the DOM [DOM] or accessibility trees may be of interest to ACT Rules that intend to test things like complexity or intention of the language. For example, an ACT Rule might test that paragraphs of text within the DOM tree do not exceed a certain readability score or **_that the text alternative of an image provides a sufficient description_**.

This means that rules that has to do with evaluating whether something is e.g. a valid text alternative should have "language" as an input aspect.

These rules are impacted, at least, as I see it: 
- [Document title is descriptive](https://act-rules.github.io/rules/c4a8a4)
- [Form field label is descriptive](https://act-rules.github.io/rules/cc0f0a)
- [Heading is descriptive](https://act-rules.github.io/rules/b49b2e)
- [No keyboard trap non-standard navigation](https://act-rules.github.io/rules/ebe86a)
- [Filename is valid accessible name](https://act-rules.github.io/rules/9eb3f6) ? 
- [Iframes with identical accessible names serve equivalent purpose](https://act-rules.github.io/rules/4b1c6c) ?
- [Links with identical accessible names serve equivalent purpose](https://act-rules.github.io/rules/b20e66)
- Most audio/video rules?

**Comment by Jym77** [comment]
Side note: we will also need to update all the examples in these rules to say that the language is English.

**Comment by annethyme** [comment]
A thought from the discussion on today's ACT Rules CG call (@WilcoFiers, especially): 
Is it important to be able to automatically detect the language of test cases when all the rules in question are either (currently considered) manual or semi-automated?

**Comment by WilcoFiers** [comment]
@annethyme I think that question may be the crux of the discussion. I would say no, checking whether or not a tester (human or machine) can accurately determine the language of a page should not be tested by the test cases for those rules. A rule that checks if the title describes the page is not about if you can determine the natural language on that page. The rule that checks if the html[lang] attribute is correct does that already.

Including the `html[lang]` attribute in these examples is a simplified way of skipping that step. Admittedly, it's not an air tight solution. Someone might ignore the lang attribute, but I'd argue that's still a reasonable enough approximation of what we're trying to communicate in the test case.

**Comment by annethyme** [comment]
@kasperisager noted that language should be an input aspect in all rules that has exeptions for characters that do not express anything in human language.

**Comment by annethyme** [comment]
As I have mentioned a few times in meetings, I think it can be dangerous to rely on the language tag to determine the language of the page.
To illustrate why, I have this example: 

## Example test case for "Link is descriptive" rule

Natural page of the page is Danish, but the language tag is set to "English" (which is wide spread issue due to using the default settings of the CMS). 

```
<html lang="en">
<head></head>
<body>
<p>Opskrifter på desserter:</p>
<ul>
<li><a href="#">Is</a></li>
</ul>
</body>
</html>
```

In English "is" = "third person singular present of be", whereas in Danish "is" = "ice cream". 
(The paragraph above the list says "Recipes for desserts:")
Evaluating the rule based on the natural language would give one (correct) result of the rule for this page, whereas evaluating the rule based on the language tag would give a different (incorrect) result of the rule.

By adding language tags to the test cases, it seems that we are encouraging implementers to rely on the language tag to determine the language, which I don't see as reliable.

**Comment by annethyme** [comment]
A way out of the above: make an assumption in all rules with Language as an input aspect, that if language tag is used to determine the language of the page, it is assumed that the language tag matches the natural language of the page.

**Comment by WilcoFiers** [comment]
Resolution from the last call was to do two things:

1. Include a `lang` attribute in the example itself
2. Include the language in the description of the test case

**Comment by Jym77** [comment]
Closed by #996.
